src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -544,22 +560,40 @@ static ADUC_Result ScriptHandler_PerformAction(const std::string& action, const
     // selected component, we will skip the 'apply' phase, and then skip the
     // remaining install-item(s).
     // Also, don't continue if WorkflowHandle is NULL in the ADUInterface_Connected->HandleStartupWorkflowData flow.
-    if (result.ResultCode == ADUC_Result_Install_Skipped_UpdateAlreadyInstalled
+    if (results.result.ResultCode == ADUC_Result_Install_Skipped_UpdateAlreadyInstalled
         || workflowData->WorkflowHandle == nullptr)
     {
         goto done;
     }
 
     aduShellArgs.emplace_back(adushconst::target_data_opt);
-    aduShellArgs.emplace_back(scriptFilePath);
+    aduShellArgs.emplace_back(results.scriptFilePath);
+    results.commandLineArgs.emplace_back(results.scriptFilePath);
+
+    // Prepare arguments based on specified api version.
+    if (apiVer == nullptr || strcmp(apiVer, "2.0") == 0)
+    {
+        std::string backcompatAction = "--action-" + action;
+        aduShellArgs.emplace_back(adushconst::target_options_opt);
+        aduShellArgs.emplace_back(backcompatAction.c_str());
+        results.commandLineArgs.emplace_back(backcompatAction.c_str());
+    }
+    else if (strcmp(apiVer, "2.1") == 0)
+    {
+        aduShellArgs.emplace_back(adushconst::target_options_opt);
+        aduShellArgs.emplace_back(HANDLER_ARG_ACTION);
+        results.commandLineArgs.emplace_back(HANDLER_ARG_ACTION);
 
-    aduShellArgs.emplace_back(adushconst::target_options_opt);
-    aduShellArgs.emplace_back(action.c_str());
+        aduShellArgs.emplace_back(adushconst::target_options_opt);
+        aduShellArgs.emplace_back(action.c_str());
+        results.commandLineArgs.emplace_back(action.c_str());
```

Comment:
maybe outside scope of this PR but it'd be really nice to export all the work data and workflow handle data that could be useful as env var for the scripts to consume without having to parse cmdline opts

Reply:
Will do this as part of  Internal Task#43162147]